### ISSUE TITLE
[Feature Request] Allow component documentation within razor file

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn#85414]
### Is your feature request related to a problem? Please describe.
It is convenient to specify both markup and code in a single `Foo.razor` file. However while it is possible to document the class' members, it's not possible to document the class itself.

For example, `MyComponent.razor`:
````cs
<markup>

@class {

  /// <summary>
  /// Some documentation.
  /// </summary>
  [Parameter] public bool Foo { get; set; }

  //...etc...
}
````

That member's documentation is recognised. But one cannot document that component itself. So we need this:

`MyComponent.razor.cs`:
````cs
/// <summary>
/// Some documentation.
/// </summary>
public partial class MyComponent : ComponentBase { }
````

### Describe the solution you'd like
To be able to document the class in the same razor file.

e.g. `MyComponent.razor`:
````cs
<markup>

/// <summary>                      <------------------ THIS
/// Some documentation.
/// </summary>
@class {

  /// <summary>                    <------------------ OR THIS (bool stripped from generated class)
  /// Some documentation.
  /// </summary>
  private bool ___Documentation;

  /// <summary>
  /// Some documentation.
  /// </summary>
  [Parameter] public bool Foo { get; set; }

  //...etc...
}
````

### Applicable Scenarios
Blazor

### Describe alternatives you've considered
A separate class. When there are many components, all those `.cs` files are just noise.

### Additional context
No.

**Comment by sh-slfcu** [comment]
Same boat. So annoying when a codebase is shared among coworkers. How do the official Microsoft components do it?

![image](https://github.com/dotnet/roslyn/assets/71898983/0bf58a2c-61b9-4905-9fba-e127ed9fb5eb)

**Comment by HugoVG** [comment]
It would be so helpful for smaller objects, I don't want to add a code behind file for a 10 line file

**Comment by Slugsie** [comment]
Yes, please add the ability to document the razor component itself without the need to add a .cs code behind file.

**Comment by AdmiralSnyder** [comment]
this would be incredibly useful for onboarding new devs into sizable codebases, too.
is this possible to get prioritized soon-ish @davidwengier 
if not, i might ask an apprentice to write a tool to create one big collective .cs file that contains lots of partial classes for our .razor files and just copies xml comment code there that maybe is put into a 
```
@attribute [XmlComment] @*<summary></summary>*@
```
line or a 
```cs
#if false XMLComment
///<summary>something</summary>
#endif
```
block
... and have that run from time to time...

**Comment by davidwengier** [comment]
I think the language feature as suggested is a non-starter. Even assuming the use of `@class` is a typo for `@code` and not a new suggested directive, the suggestion is for what is currently considered markup to have a special meaning, based on its position relative to a directive, and starting with `///`, which is not very Razor-like. I would propose something more like:

```
@documentation {
/// <summary>
/// hi there
/// </summary>
}
```

So this would mean a mulitiline directive that contains C# that is emitted before the class declaration in the generated code. We would report an error for any line that doesn't start with `///`, just so you can't sneak in an attribute :)

Thoughts, comments, objections etc?

**Comment by MarvinKlein1508** [comment]
@davidwengier that's exactly how I had it in mind from my comment back in 2023! 👍

**Comment by AdmiralSnyder** [comment]
@davidwengier i would really like to have the ability to create a oneliner at the top of the file that's _just_ the summary:
`@documentationSummary This is The Summary Text Line`
for like the 90% case
pretty please

**Comment by davidwengier** [comment]
We're not going to add two directives for documentation, but a simple directive for single line only, and if you want multiple lines then create a code-behind, seems a reasonable approach to me.

**Comment by jjonescz** [comment]
> oneliner 

you could do that with the original David's proposal if `/**` is supported too:

```razor
@documentation { /** <summary>hi there</summary> */ }
```
